src/utils/common.ts

```diff
@@ -527,3 +530,20 @@ export const toBase64Url = (data: Buffer | Uint8Array | string): string => Buffe
   .replace(/\//g, '_')
   .replace(/\+/g, '-')
   .replace(/=+$/, '');
+
+/**
+ * Get protocol by address regardless of the active network.
+ */
+export function getProtocolByAddress(address: AccountAddress) {
+  return (
+    ProtocolAdapterFactory.getAdapterByAccountAddress(address, NETWORK_TYPE_MAINNET)
+    || ProtocolAdapterFactory.getAdapterByAccountAddress(address, NETWORK_TYPE_TESTNET)
+  )?.protocol;
+}
+
+/**
+ * Generate a simple random string id.
+ */
+export function generateId() {
```

Comment:
If the address book entries would be keyed by the address or index this util can be removed.